tests/test_tokenizer.py

```diff
@@ -0,0 +1,85 @@
+import pytest
```

Comment:
Please add the shebang in the first line otherwise the script will use the system python instead of the virtual environment one.
`#!/usr/bin/env python`

Reply:
But we won't be running this file by itself. It will be found by pytest instead (I was running `python -m pytest tests` with the virtualenv activated). It shouldn't be needed here.

Reply:
pytest is also a new dependency only for development mode. We don't have any special setup instructions for dev mode yet and this will require that. 
I just checked that the standard library's unittest module also has support for parameterized test cases. I can port this to that so there's no dependency and it can be run either directly with python (with the shebang) or with the `unittest` cli.

Reply:
Pushed changes to use unittest instead.